### PR TITLE
WC: Video's page (Listing)

### PR DESCRIPTION
**Pull request by Prabhav-Sharma**
## Video Listing Page
I have added the video listing page, along with VideoCard
 and Navigation component. I've also added a VideoProvider context. 

### Other Changes
- A useDocumentTitle hook to change the document title as per the page
- A useToggle hook that can be used for states where a boolean is needed
-  A fetchVideos function to fetch videos from the database
-  Router & routes with functioning Navlinks with active and inactive styling
-  Dummy data for videos in the database
- Styling for all component and base utility and reset classes

### Important Files to Check
- [ ] src/components/Card/VideoCard/VideoCard.jsx
- [ ] src/contexts/providers/VideosProvider.js
- [ ] src/hooks/useDocumentTitle.jsx
- [ ] src/hooks/useToggle.jsx
- [ ] src/pages/Videos/Videos.jsx
- [ ] src/router/Router.jsx
- [ ] src/services/APICalls.js

P.S: Ignore other files. The high number of files changed is due to refactoring in the folder structure

Preview:
Listing page
image

Sidebar (with active link styling)
image

### CODE REVIEW

**Review by SukanyaSen530 — APPROVED** [review]
Look's Good :)

File: src/components/Card/VideoCard/VideoCard.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    author,
+    authorImg,
+    dateUploaded,
+    views,
+  } = video;
+
+  return (
+    <div className={`${styles.video_card} flex-col`}>
+      <div className={styles.top_wrapper}>
+        <img className={styles.card_img} src={thumbnail} />
```
Comment (prathmeshjagtap) [inline src/components/Card/VideoCard/VideoCard.jsx]: add alt tag to image

File: src/pages/Videos/Videos.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  authorImg:
+    "https://www.whats-on-netflix.com/wp-content/uploads/2018/09/naruto-on-netflix.jpg",
+  dateUploaded: "2 Months ago",
+  views: 167,
+};
+
+function Videos() {
+  const { videos, setVideos } = useVideos();
+
+  console.log(videos);
```
Comment (prathmeshjagtap) [inline src/pages/Videos/Videos.jsx]: remove console.log

File: src/router/Router.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      <Route path="/videos" element={<Videos />} />
+      <Route path="/video/:videoId" element={<Video />} />
+      <Route path="/history" element={<History />} />
+      <Route path="/liked" element={<Liked />} />
+      <Route path="/login" element={<Login />} />
+      <Route path="/signup" element={<SignUp />} />
+      <Route path="/watchlater" element={<WatchLater />} />
+      <Route path="/playlists" element={<Playlists />} />
+      <Route path="/playlist/:playlistID" element={<Playlist />} />
+    </Routes>
```
Comment (prathmeshjagtap) [inline src/router/Router.jsx]: you can add one more 404 page not found route

File: src/services/APICalls.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+const fetchVideos = async (setterFunc) => {
+  try {
+    const response = await axios({
+      method: "GET",
+      url: "/api/videos",
+    });
+    setterFunc(response.data.videos);
+  } catch (e) {
+    console.log("arreeeeeee", e);
```
Comment (prathmeshjagtap) [inline src/services/APICalls.js]: you can update this console.log

**Review by prathmeshjagtap — CHANGES_REQUESTED** [review]
Everything  else is good you have properly structured  your code , I liked you have used hooks for toggle and documents  title

File: src/services/APICalls.js (hunk 4 above)
Comment (Prabhav-Sharma) [reply src/services/APICalls.js]: Thanks a lot, Prathmesh, these slipped my mind.

File: src/router/Router.jsx (hunk 3 above)
Comment (Prabhav-Sharma) [reply src/router/Router.jsx]: I was going to do that later, but adding the route like the others makes more sense. Thanks. I've added it

File: src/components/Card/VideoCard/VideoCard.jsx (hunk 1 above)
Comment (Prabhav-Sharma) [reply src/components/Card/VideoCard/VideoCard.jsx]: Done

**Review by prathmeshjagtap — APPROVED** [review]

**Pull request merged**